src/manifolds/CurvedTorus.jl

```diff
@@ -0,0 +1,223 @@
+
+@doc raw"""
+    TorusInR3{TR<:Real} <: AbstractDecoratorManifold{ℝ}
+
+Surface in ℝ³ described by parametric equations:
+```math
+x(θ, φ) = (R + r\cos θ)\cos φ
+y(θ, φ) = (R + r\cos θ)\sin φ
+z(θ, φ) = r\sin θ
+```
+for θ, φ in $[-π, π)$. It is assumed that $R > r > 0$.
+
+Alternative names include anchor ring, donut and doughnut.
+
+# Constructor
+
+    TorusInR3(R, r)
+"""
+struct TorusInR3{TR<:Real} <: AbstractDecoratorManifold{ℝ}
```

Comment:
We could also call this `EmbeddedTorus`? That might be nicer if we extend that to higher dimensions at some point.

Reply:
I'm not sure it can actually be reasonably extended, I haven't seen anyone constructing similar curved tori in higher dimensions.

Reply:
Ok, I still do not like the type name `TorusinR3` – in this case there is no distinction between the one build of two circles (in the name).

Reply:
`EmbeddedTorus` seems a bit too general to me but you're right that it's better than `TorusinR3`

Reply:
I've changed the name.